crates/shared/src/request_sharing.rs

```diff
@@ -97,13 +116,26 @@ where
     }
 }
 
+#[derive(prometheus_metric_storage::MetricStorage)]
+struct Metrics {
+    /// Request sharing hits & misses
+    #[metric(labels("request_label", "result"))]
```

Comment:
I believe having one label per estimator does not give us additional information.
All requests go through the `CompetitionPriceEstimator` which sends requests to all enabled price estimators. AFAIK non-uniformly distributed metrics across different estimator could only be explained with race conditions and by the fact that `1Inch` is not configured as a native price estimator (so naturally it would see fewer requests). Both reasons don't inform us about the performance of the caching solution IMO.

Reply:
I guess I could make label optional and not record metrics for some request sharer